api/src/Validator/Period/AssertValidPeriodEndValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Validator\Period;
+
+use App\Entity\Period;
+use DateInterval;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+class AssertValidPeriodEndValidator extends ConstraintValidator {
+    public function __construct(private EntityManagerInterface $em) {
+    }
+
+    public function validate($value, Constraint $constraint) {
+        if (!$constraint instanceof AssertValidPeriodEnd) {
+            throw new UnexpectedTypeException($constraint, AssertValidPeriodEnd::class);
+        }
+
+        $period = $this->context->getObject();
+        if (!$period instanceof Period) {
+            throw new UnexpectedValueException($period, Period::class);
+        }
+
+        if ($period->scheduleEntries->count() > 0) {
+            $delta = 0;
+            if (!$period->moveScheduleEntries) {
+                $orig = $this->em->getUnitOfWork()->getOriginalEntityData($period);
+                if (null != $orig) {
+                    $delta = $orig['start']->getTimestamp() - $period->start->getTimestamp();
+                    $delta = floor($delta / 60);
+                }
+            }
+
+            // get maximal existing ScheduleEntryEnd
+            $scheduleEntryEnds = $period->scheduleEntries->map(fn ($se) => $se->periodOffset + $se->length);
```

Comment:
Ich hätte persönlich eher mit `DateTime` und `DateInterval` gearbeitet, anstatt manuell zwischen Minuten/Tagen hin- und her zu rechnen. Aber ist wohl Geschmacksache. It get's the job done.

```php
            $delta = $period->start->diff($orig['start'])
            
            $scheduleEntryEnds = $period->scheduleEntries->map(fn ($se) => $se->getEnd());
            $maxScheduleEntryEnd = max($scheduleEntryEnds->toArray())->add($delta);

            if ($maxScheduleEntryEnd > $value) {
```